projects/components/src/table/table-api.ts

```diff
@@ -4,20 +4,35 @@ import { FieldFilter, FilterValue } from '../filtering/filter/filter';
 import { FilterOperator } from '../filtering/filter/filter-operators';
 import { TableCellAlignmentType } from './cells/types/table-cell-alignment-type';
 
-export interface TableColumnConfig {
-  id: string; // This is the unique ID for the column (often same as 'name' except for composite fields)
-  name?: string; // Attribute name (for composite columns use the attribute that should be filtered/sorted)
+export interface TableColumnConfig<TableColumnOptions = unknown> {
```

Comment:
I think its safe to assume `Dictionary<unknown>` as type.

Reply:
Started off with that, but had type conflicts downstream.

Reply:
Could we do `TableColumnOptions = Dictionary<boolean>` so that we only allow a bunch of booleans?

Reply:
There can be some other cases where passing enum would also make sense for options.

Reply:
agree with possible need for enums. but why wouldn't `TableColumnOptions extends Dictionary<unknown>` type work?

Reply:
I can come around to it in some time.